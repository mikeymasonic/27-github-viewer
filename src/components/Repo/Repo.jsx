import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ name, html_url }) => (
  <ul>
    <h3>{name}</h3>
    <li><a href={html_url} target="_blank" rel="noopener noreferrer">link</a></li>
  </ul>
);

Repo.propTypes = {
  name: PropTypes.string.isRequired,
  html_url: PropTypes.string.isRequired
};

export default Repo;
