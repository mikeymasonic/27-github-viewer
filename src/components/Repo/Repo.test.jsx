import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo';

describe('Repo component', () => {
  it('renders Repo', () => {
    const wrapper = shallow(<Repo 
      name='loopr'
      html_url='https://github.com/mikeymasonic/loopr'/>);
    expect(wrapper).toMatchSnapshot();
  });
});
