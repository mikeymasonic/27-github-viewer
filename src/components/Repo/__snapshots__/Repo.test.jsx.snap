// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo component renders Repo 1`] = `
<ul>
  <h3>
    loopr
  </h3>
  <li>
    <a
      href="https://github.com/mikeymasonic/loopr"
      rel="noopener noreferrer"
      target="_blank"
    >
      link
    </a>
  </li>
</ul>
`;
